### PR TITLE
feat(auth): Failback to "none" auth if "basic" auth fails

### PR DESCRIPTION
**Pull request by ivanilves**
Solving https://github.com/ivanilves/lstags/issues/167

`docker.bintray.io` asks for basic HTTP authentication, but really requires no authentication for public images :wink: 

# GIF

### CODE REVIEW

**Comment by ivanilves** [comment]
Stress tests were broken. Cherry-picked fix from https://github.com/ivanilves/lstags/pull/166

File: tag/remote/auth/auth.go [hunk 1]
```diff
@@ -86,7 +86,14 @@ func NewToken(repo *repository.Repository, username, password string) (TokenResp
 	case "None":
 		return none.RequestToken()
 	case "Basic":
-		return basic.RequestToken(url, username, password)
+		t, err := basic.RequestToken(url, username, password)
+		if err != nil {
+			println(err.Error())
+
+			return none.RequestToken()
```
Comment (vonrabbe) [inline tag/remote/auth/auth.go]: ha ha hack!!! :smiling_imp:

**Comment by vonrabbe** [comment]
OK, let's merge this!

**Comment by ivanilves** [comment]
thanks!

**Pull request merged**